src/engine/input/key_mapping/SpecifiedInputKey.cs

```diff
@@ -57,9 +57,17 @@ public override string ToString()
 
         if (Type == InputType.Key)
         {
-            // TODO: add translation support for key names like Right, and Left etc.
-            // TODO: https://github.com/Revolutionary-Games/Thrive/issues/1895
-            text += ((KeyList)Code).ToString();
+            var key = (KeyList)Code;
+
+            text += key switch
+            {
+                KeyList.Left => TranslationServer.Translate("LEFT"),
+                KeyList.Right => TranslationServer.Translate("RIGHT"),
+                KeyList.Up => TranslationServer.Translate("UP"),
+                KeyList.Down => TranslationServer.Translate("DOWN"),
+                KeyList.Print => TranslationServer.Translate("PRINT"),
```

Comment:
While this does fix some things, it isn't enough. Every single key in KeyList should have a translation equivalent (using an editor macro to generate that text would be fine). Also it'll end up being really long so I'd prefer if that was wrapped in a static class that is put in a new file called KeyNames.cs in the key_mapping folder.